process-analyser/src/com/axonivy/solutions/process/analyser/managedbean/ProcessesAnalyticsBean.java

```diff
@@ -97,34 +97,54 @@ public Set<String> getAvailableModules() {
     return processesMap.keySet();
   }
 
+  private boolean isDiagramAndStatisticRenderable() {
+    return ObjectUtils.allNotNull(selectedProcess, selectedKpiType);
+  }
+
   public void onModuleSelect() {
-    if (StringUtils.isBlank(selectedModule)) {
-      selectedProcess = null;
+    selectedProcess = null;
+    PrimeFaces.current().ajax().update(ProcessAnalyticViewComponentId.PROCESS_DROPDOWN);
+    if (StringUtils.isNotBlank(bpmnIframeSourceUrl)) {
+      resetStatisticValue();
+      resetCustomFieldFilterValues();
     }
-    resetStatisticValue();
-    resetCustomFieldFilterValues();
+  }
+
+  private List<String> getDiagramAndStatisticComponentIds() {
+    List<String> results = new ArrayList<>();
```

Comment:
Should we return a statis list like List.of or Arrays.asList, beccause this IDs should not be changed, right?